src/TBuffer.cpp

```diff
@@ -4431,3 +4439,68 @@ void TBuffer::encodingChanged(const QString& newEncoding)
         }
     }
 }
+
+QString TBuffer::processSupportsRequest(const QString& elements)
+{
+    // strip initial SUPPORT and tokenize all of the requested elements
+    auto elementsList = elements.trimmed().remove(0, 7).split(QStringLiteral(" "), QString::SkipEmptyParts);
+    QStringList result;
+
+    auto reportEntireElement = [](auto element, auto& result) {
+        result.append("+" + element);
+
+        for (const auto& attribute : mSupportedMxpElements.value(element)) {
```

Comment:
I don't think we should report all supported arguments if only the general support is requested. The examples at least don't show this behaviour and you don't do it if a specific tag is requested either.

Reply:
MUSH seems to do it according to the [original ticket](https://github.com/Mudlet/Mudlet/issues/475) and it makes sense to me to list out all the arguments we support... I'll add listing out all arguments for a specific tag.